contracts/two-party-pol-holder/src/suite_tests/tests.rs

```diff
@@ -0,0 +1,534 @@
+fn test_instantiate_invalid_lockup_config_height_based() {
+    SuiteBuilder::default()
+        .with_lockup_config(ExpiryConfig::Block(INITIAL_BLOCK_HEIGHT - 1))
+        .build();
+}
+
+#[test]
+fn test_single_party_deposit_refund_block_based() {
+    let mut suite = SuiteBuilder::default()
+        .with_deposit_deadline(ExpiryConfig::Block(12545))
+        .build();
+
+    // party A fulfills their part of covenant but B fails to
+    let coin = suite.get_party_a_coin(Uint128::new(500));
+    suite.fund_coin(coin);
+
+    // time passes, clock ticks..
+    suite.pass_blocks(250);
+    suite.tick(CLOCK_ADDR).unwrap();
+    suite.tick(CLOCK_ADDR).unwrap();
+
+    let holder_balance = suite.get_denom_a_balance(suite.holder.to_string());
+    let router_a_balance = suite.get_denom_a_balance(suite.query_party_a().router);
+    let holder_state = suite.query_contract_state();
+
+    assert_eq!(ContractState::Complete, holder_state);
+    assert_eq!(Uint128::zero(), holder_balance);
+    assert_eq!(Uint128::new(500), router_a_balance);
+}
+
+#[test]
+fn test_single_party_deposit_refund_time_based() {
+    let current_timestamp = get_default_block_info();
+    let mut suite = SuiteBuilder::default()
+        .with_deposit_deadline(ExpiryConfig::Time(current_timestamp.time.plus_minutes(200)))
+        .build();
+
+    // party A fulfills their part of covenant but B fails to
+    let coin = suite.get_party_a_coin(Uint128::new(500));
+    suite.fund_coin(coin);
+
+    // time passes, clock ticks..
+    suite.pass_minutes(250);
+    suite.tick(CLOCK_ADDR).unwrap();
+    suite.tick(CLOCK_ADDR).unwrap();
+
+    let holder_balance = suite.get_denom_a_balance(suite.holder.to_string());
+    let router_a_balance = suite.get_denom_a_balance(suite.query_party_a().router);
+    let holder_state = suite.query_contract_state();
+
+    assert_eq!(ContractState::Complete, holder_state);
+    assert_eq!(Uint128::zero(), holder_balance);
+    assert_eq!(Uint128::new(500), router_a_balance);
+}
+
+#[test]
+fn test_single_party_deposit_refund_no_deposit_deadline() {
+    let mut suite = SuiteBuilder::default().build();
+
+    // party A fulfills their part of covenant but B fails to
+    let coin = suite.get_party_a_coin(Uint128::new(500));
+    suite.fund_coin(coin);
+
+    // time passes, clock ticks..
+    suite.pass_minutes(25000000);
+    suite.tick(CLOCK_ADDR);
+    suite.tick(CLOCK_ADDR);
+    let resp: ContractError = suite.tick(CLOCK_ADDR).unwrap_err().downcast().unwrap();
+
+    // we assert that holder still holds the tokens and did not advance the state
+    let holder_balance = suite.get_denom_a_balance(suite.holder.to_string());
+    let holder_state = suite.query_contract_state();
+
+    assert_eq!(ContractState::Instantiated, holder_state);
+    assert_eq!(Uint128::new(500), holder_balance);
+    assert_eq!(ContractError::InsufficientDeposits {}, resp);
+}
+
+#[test]
+fn test_holder_active_does_not_allow_claims() {
+    // unimplemented!()
+}
```

Comment:
You can put the `#[ignore]` tag, even give a reason `#[ignore = "reason"]`, it shows it when you run `cargo test`